agent/ecscni/netconfig_windows.go

```diff
@@ -31,6 +41,14 @@ func NewVPCENIPluginConfigForTaskNSSetup(eni *eni.ENI, cfg *Config) (*libcni.Net
 		Nameservers: cfg.InstanceENIDNSServerList,
 	}
 
+	// Validate MAC Address, ENI IP Address and ENI Gateway address used for CNI plugin configuration.
+	// Other params are generated at runtime and are considered safe.
+	if !isValid(eni.MacAddress) || !isValid(eni.GetPrimaryIPv4AddressWithPrefixLength()) ||
+		!isValid(eni.GetSubnetGatewayIPv4Address()) {
+		return nil, errors.New("failed to create vpc-eni plugin configuration for setting up " +
```

Comment:
nit: Most of this text should be part of a log message and the error should just be something like `failed data validation`